src/main/java/com/joy/joyfillandroid/TextField.kt

```diff
@@ -0,0 +1,175 @@
+package com.joy.joyfillandroid
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Typeface
+import android.os.Build
+import android.view.Gravity
+import android.widget.LinearLayout
+import androidx.appcompat.widget.AppCompatTextView
+
+class TextField(context: Context?) : AppCompatTextView(context!!) {
+
+    private var borderColor: Int = 0
+    private var cornerRadius: Int = 0
+    private var borderWidth: Int = 0
+    private var isBold: Boolean = false
+    private var isItalic: Boolean = false
+    private var fontName: String? = null
+
+    companion object {
+        private const val DEFAULT_TEXT_SIZE_SP = 6f
+        private const val DEFAULT_TEXT_COLOR = android.R.color.black
+    }
+
+    init {
+        // Read attributes from XML
+
+        text = "Text Field"
+        textSize = TextField.DEFAULT_TEXT_SIZE_SP
+        val linearLayout = LinearLayout(context)
+        val layoutParams = LinearLayout.LayoutParams(
+            LinearLayout.LayoutParams.WRAP_CONTENT,
+            LinearLayout.LayoutParams.WRAP_CONTENT
+        )
+        linearLayout.orientation = LinearLayout.VERTICAL
+        linearLayout.gravity = Gravity.CENTER_VERTICAL
+        linearLayout.layoutParams = layoutParams
+        layoutParams.setMargins(10,10,10,10)
+
+        setTextColor(Color.BLACK)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            typeface = Typeface.create(null, 500, false)
+        }
+        val padding = resources.getDimensionPixelSize(R.dimen.custom_edittext_padding)
+        setPadding(16, 0, 0, 10)
+
+        // Set default text color
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
Why would the build version matter for setting the default text color?

Reply:
This condition is used to check the Android version of the device at runtime. It specifically checks if the device's SDK version is equal to or greater than Android Marshmallow.
{
this is particularly useful when dealing with features or APIs that are available only on newer Android versions.
}